src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/DisallowedTypeConverterFactory.cs

```diff
@@ -10,8 +10,7 @@ internal sealed class DisallowedTypeConverterFactory : JsonConverterFactory
     {
         public override bool CanConvert(Type type)
         {
-            // If a value type is added, also add a test that
-            // shows NSE is thrown when Nullable<T> is (de)serialized.
+            // If a type is added, also add to the souce-gen project.
```

Comment:
BTW, that message is maybe weasel words? eg "since they can lead to security issues" --> "because it is insecure"

Reply:
Agree with Layomi we should remove references to security from the exception message, since this class has evolved into a general-purpose converter for unsupported types. We might consider special casing the error message for `System.Type` and `SerializationInfo` (e.g. by injecting a `reason` string to the constructed converter) but this is likely overkill.

Reply:
Removing the last section works for me:

from

"Serialization and deserialization of '{0}' instances are not supported and should be avoided since they can lead to security issues."

to

"Serialization and deserialization of '{0}' instances are not supported."

Reply:
> We might consider special casing the error message for System.Type and SerializationInfo (e.g. by injecting a reason string to the constructed converter) but this is likely overkill.

FWIW there was a discussion on how to scale these exception messages (https://github.com/dotnet/runtime/pull/42605#issuecomment-697049134, https://github.com/dotnet/runtime/pull/42605#issuecomment-697874991). Another approach to consider in the future is to add a `JsonTypeDisallowReason` enum which is passed to a new converter ctor overload, which would determine what exception message is shown.

Reply:
> add a JsonTypeDisallowReason enum 

Unless there is a requirement for the "security issue" in the message, I think this is overkill. It would also require 2 converter instances and add code\tests to separate the unsupported types into "security" and "non security".

Reply:
cc @GrabYourPitchforks